sklearn/utils/_metadata_requests.py

```diff
@@ -177,6 +179,21 @@ def requests(self):
         """Dictionary of the form: ``{key: alias}``."""
         return self._requests
 
+    def _assume_requested(self, router):
+        """Convert all ERROR_IF_PASSED to REQUESTED.
```

Comment:
Fair points. I'm going to see if I can find a solution which is not too convoluted. I wish we could do version 2 and break all of these rather than being backward compatible with such a change 😁

Reply:
It's also okay at this point if you don't try to do something generic. We've factored out a lot of the work already. We can just get something out there that offers the API we want.... Yes, not needing deprecation makes this a lot simpler!

Reply:
I found a much better way to do this, and indeed your point was even valid for `Multioutput*`, since `fit` already forwards everything but `partial_fit` only forwards `sample_weight`.